bin/civi-download-tools

```diff
@@ -22,7 +22,7 @@ LOCKFILE="$TMPDIR/civi-download-tools.lock"
 LOCKTIMEOUT=90
 HUB_VERSION="2.14.2"
 JOOMLATOOLS_CONSOLE_VERSION="v1.5.9"
-COMPOSER_VERSION="2.2.18"
+COMPOSER_VERSION="2.3.10"
```

Comment:
My main hesitation with merging was this -- for some D10 scenarios (e.g. installing directly from `drupal.git`), you need composer@2.3, but this breaks support for some older versions of Civi.

After testing that a bit, the breaking-point appears to be Civi 5.47 -- i.e. composer@2.3.10 can install civicrm@5.48 but not civicrm@5.47.

Discussed a bit with Seamus on MM, and we both felt that was acceptable -- it's good for us to support D10 in more CI scenarios, and you can still install the past 2 ESR's (5.51 and 5.57).